### DISCUSSION TITLE
Local QDrant data

### DISCUSSION THREAD

**Discussion by heiko-braun** [heiko-braun/cai#14] [Ideas]
How can we get that data out of qdrant and make it available locally?
(In relation to https://github.com/heiko-braun/cai/issues/13)

**Comment by heiko-braun** [comment]
We might be able to leverage the snapshot files to create a local qdrant instance. If the snapshot files are not usable out-of-the-box, we might have to go through an import:
1. create local qdrant
2. import from snapshot as documented in https://qdrant.tech/documentation/tutorials/create-snapshot/

If you want to try, here is the API documentation for the snapshot import : https://qdrant.github.io/qdrant/redoc/index.html?v=master#tag/collections/operation/recover_from_snapshot

**Reply by lordrip** [reply]
I just checked the docs and it seems pretty straightforward, I'll try it with my instance first.

That being said, should we aim for having "fixed" collection names? I see we have some collections already, and some with a `_2` prefix, like `tooling_guide` and `tooling_guide_2`, which ones should we aim for?

**Reply by heiko-braun** [reply]
These are merely iterations where I tried different storage formats. The ones that matter are these:

- tooling_guide_2
- spring_reference_2
- spring_get_started_2
- quarkus_getting_started_2
- quarkus_reference_2
- rhaetor.github.io_components_2
- rhaetor.github.io_2

We can ignore the remainder. And yes, if we touch this, we can cleanup the names. But the then you also need to adjust `CustomTools.py` and `agent.py`.

**Comment by lordrip** [comment]
Now that https://github.com/heiko-braun/cai/pull/17 is merged, we need the upload command, I'll try to tackle it soon

**Comment by heiko-braun** [comment]
# Thoughts on packaging and deployment of the vectorstore

It would be lovely to have the actually collections data in this repo and simply run an in-memory QDrant over it. That would allow me to continue to use github workspaces for development.

With a baseline like this, it would be easy to build a self-contained container image that we can leverage in prod or as an alternative for people who want to use docker compose or something similar for development.

What do you think?

**Reply by lordrip** [reply]
I've downloaded the `camel-quarkus` collection and it was 65Mb... I wonder if this might be a deal breaker for GitHub workspaces.
